### ISSUE TITLE
Marking the input `version` as `@Nullable` is incorrect.

### ISSUE THREAD

**Issue by piotrooo** [semver4j/semver4j#184]
If you pass in `null` to this method, it throws a `NullPointerException` as it fails to create a regex matcher:
```
java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null
	at java.base/java.util.regex.Matcher.getTextLength(Matcher.java:1769)
	at java.base/java.util.regex.Matcher.reset(Matcher.java:415)
	at java.base/java.util.regex.Matcher.<init>(Matcher.java:252)
	at java.base/java.util.regex.Pattern.matcher(Pattern.java:1134)
	at org.semver4j.internal.Coerce.coerce(Coerce.java:20)
	at org.semver4j.Semver.coerce(Semver.java:89)
```

_Originally posted by @NielsDoucet in https://github.com/semver4j/semver4j/pull/177#discussion_r1345639824_

**Comment by DIG-** [comment]
There is two ways to resolve:
- Mark param as `NotNull`
- Return null if null

**Comment by piotrooo** [comment]
I thought about returning `null` when parameter is `null`. There is no possibility for coercing version on nullable - it's impossible.

**Comment by NielsDoucet** [comment]
https://github.com/semver4j/semver4j/blob/main/src/test/java/org/semver4j/SemverTest.java#L743
I also noticed that there wasn't a test for these edge cases (like `null` and an empty string).

**Comment by piotrooo** [comment]
This's a major oversight...

**Comment by piotrooo** [comment]
Fixed in [v5.2.2](https://github.com/semver4j/semver4j/releases/tag/v5.2.2)
